compiler/codegen/expr.cpp

```diff
@@ -1810,7 +1810,7 @@ GenRet codegenTernary(GenRet cond, GenRet ifTrue, GenRet ifFalse)
 #endif
 
   if( info->cfile ) {
-    ret.c = "(" + cond.c + ")?(" + ifTrue.c + "):(" + ifFalse.c + ")";
+    ret.c = "( (" + cond.c + ")?(" + ifTrue.c + "):(" + ifFalse.c + ") )";
   } else {
```

Comment:
I included this under the `catch` implementation section, since it actually inhibited the proper function of `catch`.